### PR TITLE
Transition other Travis jobs to Github Actions

### PR DESCRIPTION
**Pull request by PGijsbers**
I think this covers the previous work that was done by Travis jobs:
 - build docs, deploy docs (when on a push to `develop`  and `master`)
 - check the distribution
 - run the unit tests, check no file left behind
 - report coverage (accidentally disabled before the hackathon)
 - pre-commit check
 - PEP 561 compliance check

I opted to split workflows where Travis branched merely on conditionals. I think this makes each individual workflow easier to understand. There is probably some improvement to be made by using predefined Github Actions, but I think this is good enough for me right now. I would rather focus on setting up Github Actions for our Windows testing.

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        pip install -e .[examples,examples_unix]
+        pip install sphinx-gallery sphinx_bootstrap_theme numpydoc
```
Comment (mfeurer) [inline .github/workflows/docs.yaml:16]: Could we make these extra requirements for documentation building?

File: .github/workflows/dist.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Twine check
+      run: |
+        pip install twine
+        last_dist=$(ls -t dist/openml-*.tar.gz | head -n 1)
+        twine check $last_dist
+    - name: Install dist
+      run: |
+        last_dist=$(ls -t dist/openml-*.tar.gz | head -n 1)
+        pip install $last_dist
+    - name: PEP 561 Compliance
```
Comment (mfeurer) [inline .github/workflows/dist.yaml:26]: I'm not 100% sure if this test is sufficient. We only test if it's installable, but not whether the installed package actually works. Having said that, as we don't compile anything, it's maybe sufficient.

**Review by mfeurer — COMMENTED** [review]
This looks great, just one minor change request. 

I think it's good to have every test as a separate file, it's much cleaner now like this.

File: .github/workflows/dist.yaml (hunk 2 above)
Comment (PGijsbers) [reply .github/workflows/dist.yaml:26]: As far as I can tell, unless we're changing something about the structure of the package (i.e. include more than just `openml`), I don't think results should differ. To keep concerns separate and prevent additional load on the server, I decided not to run an additional test suite here. I'd say we wait until we encounter a problem from this before we add it.

Comment (mfeurer) [reply .github/workflows/dist.yaml:26]: Makes sense.

**Review by mfeurer — APPROVED** [review]

**Comment by joaquinvanschoren** [comment]
@all-contributors please add @pgijsbers for test

**Pull request merged**